src/exasolvs/queryrenderer/ScalarFunctionAppender.lua

```diff
@@ -0,0 +1,313 @@
+local ExpressionAppender = require("exasolvs.queryrenderer.ExpressionAppender")
+local AbstractQueryAppender = require("exasolvs.queryrenderer.AbstractQueryAppender")
+
+--- Appender for scalar functions in an SQL statement.
+-- @classmod ScalarFunctionAppender
+local ScalarFunctionAppender = {}
+ScalarFunctionAppender.__index = ScalarFunctionAppender
+setmetatable(ScalarFunctionAppender, {__index = AbstractQueryAppender})
+
+--- Create a new instance of a `ScalarFunctionRenderer`.
+-- @param out_query query to which the function will be appended
+-- @return renderer for scalar functions
+function ScalarFunctionAppender:new(out_query)
+    assert(out_query ~= nil,
+            "Renderer for scalar function requires a query object that it can append to.")
```

Comment:
Could you avoid duplication by moving this check to `AbstractQueryAppender._init(self, out_query)`?

Reply:
I could, but I want the program to terminate as early as possible. This is better pinpointing in my eyes.